go-controller/pkg/ovn/ovn.go

```diff
@@ -59,20 +57,57 @@ type ACLLoggingLevels struct {
 	Deny  string `json:"deny,omitempty"`
 }
 
-// Controller structure is the object which holds the controls for starting
-// and reacting upon the watched resources (e.g. pods, endpoints)
-type Controller struct {
+// ControllerConnection structure is place holder for all fields shared among controllers.
+type ControllerConnection struct {
```

Comment:
there are two levels of sharing, one is ControllerConnection(), and another is controllerInfo (see https://github.com/cathy-zhou/ovn-kubernetes/blob/multi-network-mr8/go-controller/pkg/ovn/ovn.go#L107), the former is info shared by all controllers, and the latter added some important information per-controller (netName, Prefix, etc.) which will be used for sharing code between controllers (again, see example in https://github.com/cathy-zhou/ovn-kubernetes/blob/multi-network-mr8/go-controller/pkg/ovn/master.go#L203). 

which level should be named as BaseNetworkController?

Reply:
controllerInfo is no longer in this PR but it was before, and ControllerConnection was not here when I reviewed this PR for the first time but it now contains the same stuff that controllerInfo had when I first saw it, so I am deeply confused. 

In any case, I don't see why we need two different types for this, given that ControllerConnection is created per controller, at least on this PR. And I see it fitting as a `BaseNetworkController`.

Reply:
it is just naming. ControllerConnection meant to be Network independent (no specific per-network information), so it could be used elsewhere. (for example, by controllerManager module).

controllerInfo has additional per-network information.

In this PR, since secondary network controller is not yet introduced, there is no per-controller information yet.

I could rename controllerInfo  as BaseNetworkController if you prefer.

Reply:
I would rather have that the current ControllerConnection as BaseNetworkController now and then we can change it when you introduce controllerInfo if needed.
If later on controllerInfo does not fit into BaseNetworkController, we can have some other name like NetworkControlerInfo.